cmd/tempo/app/modules.go

```diff
@@ -138,16 +139,21 @@ func (t *App) initIngester() (services.Service, error) {
 }
 
 func (t *App) initQuerier() (services.Service, error) {
+
 	// validate worker config
-	// if we're not in single binary mode and worker address is not specified - bail
-	if t.cfg.Target != All && t.cfg.Querier.Worker.FrontendAddress == "" {
-		return nil, fmt.Errorf("frontend worker address not specified")
-	} else if t.cfg.Target == All {
+	switch t.cfg.Target {
```

Comment:
no reason to change this. we could probably improve the comment since there's a new mode, but the logic is fine.

Reply:
I agree, I was mostly exploring.